### PR TITLE
Union topics are not supported in Java

### PR DESCRIPTION
**Pull request by jrw972**
Problem
-------

Union topics do not compile with Java.

Solution
--------

Add support and a extend a test for union topics in Java.

### CODE REVIEW

File: dds/idl/ts_generator.cpp [hunk 1]
```diff
@@ -705,6 +705,15 @@ namespace java_ts_generator {
       "}\n\n";
   }
 
+  /// called directly by dds_visitor::visit_structure() if -Wb,java
+  void generate(AST_Structure* node) {
+    generate_common(node);
+  }
+
+  /// called directly by dds_visitor::visit_structure() if -Wb,java
```
Comment (mitza-oci) [inline dds/idl/ts_generator.cpp:713]:
```suggestion
  /// called directly by dds_visitor::visit_union() if -Wb,java
```

Comment (mitza-oci) [reply dds/idl/ts_generator.cpp:713]: Also I suppose the `directly` in this comment and the one above no longer apply

**Review by mitza-oci — APPROVED** [review]

File: dds/idl/marshal_generator.cpp [hunk 2]
```diff
@@ -1721,6 +1725,64 @@ namespace {
...
+        string() +
+        field_type_name(dynamic_cast<AST_Field*>(field), type) + " temp = " + initializer + ";\n" +
+        "uni.value." + field->local_name()->get_string() + "(temp);\n";
+    } else {
+      return
+        string() +
+        field_type_name(dynamic_cast<AST_Field*>(field), type) + " temp;\n" +
+        "OpenDDS::DCPS::set_default(temp);\n" +
+        "uni.value." + field->local_name()->get_string() + "(temp);\n";
+    }
```
Comment (iguessthislldo) [inline dds/idl/marshal_generator.cpp:1783]: This looks like should be handled by `type_to_default`.

**Review by iguessthislldo — CHANGES_REQUESTED** [review]

File: dds/idl/marshal_generator.cpp [hunk 3]
```diff
@@ -1721,6 +1725,17 @@ namespace {
     }
   }
 
+  string initializeUnion(const std::string& indent, AST_Decl* field, const string& /*name*/,
+                         AST_Type* type, const string& /*prefix*/, bool /*wrap_nested_key_only*/, Intro& /*intro*/,
+                         const string& /*stru*/)
+  {
+    return
+      string() +
```
Comment (iguessthislldo) [inline dds/idl/marshal_generator.cpp:1733]: `field_type_name` returns a `std::string`, so is this needed?

**Review by iguessthislldo — APPROVED** [review]

**Pull request merged**